lib/streamlit/elements/widgets/options_selector/button_group.py

```diff
@@ -0,0 +1,314 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable, Literal, Sequence, cast, get_args
+
+from streamlit.elements.form import current_form_id
+from streamlit.elements.lib.utils import (
+    maybe_coerce_enum_sequence,
+)
+from streamlit.elements.widgets.options_selector.feedback_utils import (
+    FeedbackOptions,
+    FeedbackSerde,
+    get_mapped_options,
+)
+from streamlit.elements.widgets.options_selector.options_selector_utils import (
+    MultiSelectSerde,
+    check_multiselect_policies,
+    ensure_indexable_and_comparable,
+    get_default_indices,
+)
+from streamlit.errors import StreamlitAPIException
+from streamlit.proto.ButtonGroup_pb2 import ButtonGroup as ButtonGroupProto
+from streamlit.runtime.metrics_util import gather_metrics
+from streamlit.runtime.scriptrunner import get_script_run_ctx
+from streamlit.runtime.state import register_widget
+from streamlit.runtime.state.common import (
+    RegisterWidgetResult,
+    WidgetDeserializer,
+    WidgetSerializer,
+    compute_widget_id,
+    save_for_app_testing,
+)
+from streamlit.type_util import Key, OptionSequence, T, V, to_key
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+    from streamlit.runtime.state import (
+        WidgetArgs,
+        WidgetCallback,
+        WidgetKwargs,
+    )
+
+
+def _build_proto(
+    widget_id: str,
+    formatted_options: Sequence[ButtonGroupProto.Option],
+    default_values: list[int],
+    disabled: bool,
+    current_form_id: str,
+    click_mode: ButtonGroupProto.ClickMode.ValueType,
+    selection_visualization: ButtonGroupProto.SelectionVisualization.ValueType = (
+        ButtonGroupProto.SelectionVisualization.ONLY_SELECTED
+    ),
+) -> ButtonGroupProto:
+    proto = ButtonGroupProto()
+
+    proto.id = widget_id
+    proto.default[:] = default_values
+    proto.form_id = current_form_id
+    proto.disabled = disabled
+    proto.click_mode = click_mode
+
+    for formatted_option in formatted_options:
+        proto.options.append(formatted_option)
+    proto.selection_visualization = selection_visualization
+    return proto
+
+
+class ButtonGroupMixin:
+    @gather_metrics("feedback")
+    def feedback(
+        self,
+        options: FeedbackOptions = "thumbs",
+        *,
+        key: str | None = None,
+        disabled: bool = False,
+        on_change: WidgetCallback | None = None,
+        args: Any | None = None,
+        kwargs: Any | None = None,
```

Comment:
Why do we add this? It just seems like unnecessary tech debt without any reasonable benefit. We deliberately didn't add it when adding selections to charts & dataframe.

Reply:
I totally agree, but it was blocking feedback from @jrieke to keep the different APIs aligned.